src/qibolab/instruments/emulator/pulse_simulator.py

```diff
@@ -180,8 +181,70 @@ def play(
             "output_states": output_states,
         }
 
+        results["demodulation"] = {}
+
+        if execution_parameters.acquisition_type is AcquisitionType.DISCRIMINATION:
+            if execution_parameters.readout_simulator_config:
+                for index, ro_pulse in enumerate(ro_pulse_list):
+                    # obtain qubit with its qubitID
+                    qubit = qubits[self.get_qubit(ro_pulse.qubit, qubits)]
+                    # bare_resonator_frequency is prepared considering readout pulse frequency = lambshifted readout pulse frequency,
+                    # such that the V_I/V_Q are maximally seperated (i.e.: bare_resonator_frequency = lambshifted readout pulse frequency + lamb shift)
+                    # solve quadratic equation for bare_resonator frequency
+                    g = execution_parameters.readout_simulator_config["g"]
+                    noise_model = execution_parameters.readout_simulator_config[
+                        "noise_model"
+                    ]
+                    internal_Q = execution_parameters.readout_simulator_config[
+                        "internal_Q"
+                    ]
+                    coupling_Q = execution_parameters.readout_simulator_config[
+                        "coupling_Q"
+                    ]
+                    sampling_rate = execution_parameters.readout_simulator_config[
+                        "sampling_rate"
+                    ]
+                    readout = ReadoutSimulator(
+                        qubit=qubit,
+                        g=g,
+                        noise_model=noise_model,
+                        internal_Q=internal_Q,
+                        coupling_Q=coupling_Q,
+                        sampling_rate=sampling_rate,
+                    )
```

Comment:
You can expand the dictionary as keyword-argument pairs
```suggestion
                    readout = ReadoutSimulator(
                        **execution_parameters.readout_simulator_config
                    )
```